internal/provider/incident_user_data_source.go

```diff
@@ -0,0 +1,207 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/incident-io/terraform-provider-incident/internal/apischema"
+	"github.com/incident-io/terraform-provider-incident/internal/client"
+	"github.com/samber/lo"
+)
+
+var (
+	_ datasource.DataSource              = &IncidentUserDataSource{}
+	_ datasource.DataSourceWithConfigure = &IncidentUserDataSource{}
+)
+
+func NewIncidentUserDataSource() datasource.DataSource {
+	return &IncidentUserDataSource{}
+}
+
+type IncidentUserDataSource struct {
+	client *client.ClientWithResponses
+}
+
+type IncidentUserDataSourceModel struct {
+	BaseRole    *RBACRole    `tfsdk:"base_role" json:"base_role"`
+	CustomRoles []*RBACRole  `tfsdk:"custom_roles" json:"custom_roles"`
+	Email       types.String `tfsdk:"email" json:"email"`
+	ID          types.String `tfsdk:"id" json:"id"`
+	Name        types.String `tfsdk:"name" json:"name"`
+	Role        types.String `tfsdk:"role" json:"role"`
+	SlackUserId types.String `tfsdk:"slack_user_id" json:"slack_user_id"`
+}
+
+type IncidentUserRequest struct {
+	ID types.String `tfsdk:"id"`
+}
+
+type RBACRole struct {
+	Description types.String `tfsdk:"description"`
+	Id          types.String `tfsdk:"id"`
```

Comment:
```suggestion
	ID          types.String `tfsdk:"id"`
```
the inconsistency is killing me